### PR TITLE
Add Speech API quickstart sample.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: speech/cloud-client/pom.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<!--
+  Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline speech/cloud-client/pom.xml]: No need for "All Rights Reserved."

File: speech/cloud-client/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Install [Maven](http://maven.apache.org/).
+
+Build your project with:
+
+  mvn clean package -DskipTests
+
+You can then run a given `ClassName` via:
+
+  mvn exec:java -Dexec.mainClass=com.example.speech.ClassName \
+      -DpropertyName=propertyValue \
```
Comment (lesv) [inline speech/cloud-client/README.md]: I don't see the need for any arguments in your sample.

Comment (jmdobry) [inline speech/cloud-client/README.md]: Done.

File: speech/cloud-client/pom.xml (hunk 1 above)
Comment (jmdobry) [inline speech/cloud-client/pom.xml]: Done.

File: speech/cloud-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<!--
```
Comment (jerjou) [inline speech/cloud-client/pom.xml]: Need to add this to the top-level pom.xml in order for the tests to run

Comment (jmdobry) [inline speech/cloud-client/pom.xml]: Really? The test definitely ran during the [travis build](https://travis-ci.org/GoogleCloudPlatform/java-docs-samples/builds/197720320), or are you refereing to something else?

Comment (jerjou) [inline speech/cloud-client/pom.xml]: Huh.
...okay, yeah - it's a quirk of how we run travis builds. ie for PRs, we don't run the full suite of tests - we see what files have changed, then cd into that directory before running tests. So for the PR, it'll run the right tests, but when running the full suite, it won't know about these tests without being in the parent pom.

**Pull request merged**